catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/converter/KmlToJtsCoordinateConverter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.spatial.kml.converter;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections.CollectionUtils;
+
+public class KmlToJtsCoordinateConverter {
+
+    private KmlToJtsCoordinateConverter() {
+
+    }
+
+    public static com.vividsolutions.jts.geom.Coordinate from(
+            de.micromata.opengis.kml.v_2_2_0.Coordinate kmlCoordinate) {
+        if (kmlCoordinate == null) {
+            return null;
+        }
+
+        return new com.vividsolutions.jts.geom.Coordinate(kmlCoordinate.getLongitude(),
+                kmlCoordinate.getLatitude(),
+                kmlCoordinate.getAltitude());
+    }
+
+    public static com.vividsolutions.jts.geom.Coordinate[] from(
+            List<de.micromata.opengis.kml.v_2_2_0.Coordinate> kmlCoordinates) {
+        if (CollectionUtils.isEmpty(kmlCoordinates)) {
+            return null;
+        }
+
+        List<com.vividsolutions.jts.geom.Coordinate> jtsCoordinates = kmlCoordinates.stream()
+                .filter(Objects::nonNull)
+                .map(KmlToJtsCoordinateConverter::from)
+                .collect(Collectors.toList());
+        if (CollectionUtils.isEmpty(jtsCoordinates)) {
+            return null;
+        }
+
+        return jtsCoordinates.toArray(new com.vividsolutions.jts.geom.Coordinate[jtsCoordinates.size()]);
```

Comment:
I think it's more performant to use toArray with a size 0 array

Reply:
Changed